### PR TITLE
Add Java 22 support

### LINKED ISSUE #25562: JHipster 8 doesn't work with Java 22
##### **Overview of the issue**

JHipster 8 doesn't work with Java 22. If you're using Gradle, you get an error:

> BUG! exception in phase 'semantic analysis' in source unit 'precompiled_…' Unsupported class file major version 66

If you're using Maven, `CRLFLogConverterTest` doesn't pass.

Screenshot 2024-03-20 at 9 06 59 AM

##### **Motivation for or Use Case**

We should support the latest release of Java.

##### **Reproduce the error**

```
sdk install java open-22
jhipster --defaults
```

##### **Suggest a Fix**

Upgrade to Gradle 8.7 and fix the test. 

##### **JHipster Version(s)**

8.2.0

### PR DESCRIPTION
**Pull request by mraible**
Fixes #25562.

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by mraible** [comment]
I think it'd be good to change our build process to test Java 17 and 22. Not sure how to do that.

**Comment by mraible** [comment]
It looks like Gradle 8.7 only supports building with Java 22, not running with it. From https://docs.gradle.org/current/release-notes.html#support-for-building-projects-with-java-22:

> Gradle now supports [using Java 22](https://docs.gradle.org/current/userguide/compatibility.html#java) for compiling, testing, and starting other Java programs. Selecting a language version is done using [toolchains](https://docs.gradle.org/current/userguide/toolchains.html).
>
> You cannot run Gradle 8.7 itself with Java 22 because Groovy still needs to support JDK 22. However, future versions are expected to provide this support.

File: test-integration/workflow-samples/angular.json [hunk 1]
```diff
@@ -25,7 +25,7 @@
       "app-sample": "ng-mongodb-kafka-cucumber",
       "entity": "mongodb",
       "environment": "dev",
-      "java": "21"
```
Comment (OmarHawk) [inline test-integration/workflow-samples/angular.json:28]: 21 would be still kind of interesting, as it is an LTS supported version, just like jdk 17

**Review by DanielFran — APPROVED** [review]

**Comment by tmetn** [comment]
I can confirm the issue on Java 21 also, have tried on different gradle versions -current 8.8-, and I'm at 8.6.0 on jhipster. I thought it was related somehow to sonarqube, though removing the plugin was not helpful.

**Pull request merged**